src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,99 @@
+package duke.parser;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.commands.Command;
+import duke.commands.DeadlineCommand;
+import duke.commands.DeleteCommand;
+import duke.commands.EventCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.InvalidCommand;
+import duke.commands.ListCommand;
+import duke.commands.MarkCommand;
+import duke.commands.TasksCommand;
+import duke.commands.TodoCommand;
+import duke.commands.UnmarkCommand;
+import duke.data.exception.DukeException;
+
+/**
+ * This class encapsulates the Parser
+ */
+public class Parser {
+
+    /**
+     * Checks if the input is valid
```

Comment:
Perhaps this method is more like making sense of user input, while the method below checks for validity of input. 🤔